File: queue/Java/Queue.java
public class Queue {
	private int[] values;
	private int start;
	private int end;
	private int total;

	public Queue() {
		values = new int[10];
		start = 0;
		end = 0;
		total = 0;
	}

	public void insert(int e) {
		values[end] = e;
		end = (end + 1)%10;
		total++;
	}

	public int remove() {
		int element = values[start];
		start = (start + 1) % 10;
		total--;
		return element;
	}

	public boolean isEmpty() {
		return (total == 0);
	}

	public boolean isFull() {
		return (total == 10);
	}
}

File: queue/C/queue.c
#include "queue.h"

void create(struct Queue *q) {
    q->start = 0;
    q->end = 0;
    q->total = 0;
}

void insertItem(struct Queue *q, int element) {
    q->values[q->end] = element;
    q->end = (q->end + 1) % 10;
    q->total++;
}

int removeItem(struct Queue *q) {
    int element = q->values[q->start];
    q->start = (q->start + 1) % 10;
    q->total--;
    return element;
}

int isEmpty(struct Queue q) {
    return (q.total == 0);
}

int isFull(struct Queue q) {
    return (q.total == 10);
}